flang/lib/Semantics/expression.cpp

```diff
@@ -2167,15 +2167,15 @@ MaybeExpr ExpressionAnalyzer::Analyze(
           result.Add(*symbol, Fold(std::move(*value)));
           continue;
         }
-        if (IsNullPointer(*value)) {
+        if (IsNullPointer(&*value)) {
```

Comment:
What is the "&*" doing here? I am inclined to read this as dereference the pointer and then find the address of the value stored at the pointer you just dereferenced which I would think be the identity functions on pointers, but this must not be what this is doing.